two_encoder_module/two_encoder_src/two_encoder_src.ino

```diff
@@ -109,18 +115,17 @@ struct Encoder
     uint8_t pinB;
     uint8_t buttonPin;
     uint8_t buttonState;
-    int pinAPrevious;
-    int pinBPrevious;
+    uint8_t pinAPrevious;
+    uint8_t pinBPrevious;
```

Comment:
You are right that digitalRead returns an int, but it is not a problem to cast it on uint8_t, because the return value is only HIGH or LOW. Only analogRead function gives you a real 16bit value back. It is one of the very stupid things in the Arduino framework and a reason why the digitalRead and digitalWrite function is so slow.